### PR TITLE
[HttpKernel] Better exception page when the invokable controller returns nothing 

### PR DESCRIPTION
**Pull request by dimabory**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no    
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #27138
| License       | MIT
| Doc PR        | 

---

__Prerequisites__
_Configure invokable controller_
```php
# config/routes.yaml
index:
    path: /
    controller: App\Controller\Start
```

__Before:__
before

__After:__
after

---

Take a look for an enhancement/refactoring in `HttpKernel.php`

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Exception/ControllerDoesNotReturnResponseException.php [hunk 1]
```diff
@@ -67,9 +67,15 @@ private function parseControllerDefinition(callable $controller): ?array
         if (\is_object($controller)) {
             $r = new \ReflectionClass($controller);
 
+            try {
+                $line = $r->getMethod('__invoke');
```
Comment (ro0NL) [inline src/Symfony/Component/HttpKernel/Exception/ControllerDoesNotReturnResponseException.php]: :thinking:

Comment (dimabory) [reply src/Symfony/Component/HttpKernel/Exception/ControllerDoesNotReturnResponseException.php]: Yeah, missed `->getEndLine()`. I will change it asap.

Comment (dimabory) [reply src/Symfony/Component/HttpKernel/Exception/ControllerDoesNotReturnResponseException.php]: Done

File: src/Symfony/Component/HttpKernel/HttpKernel.php [hunk 2]
```diff
@@ -154,18 +154,17 @@ private function handleRaw(Request $request, int $type = self::MASTER_REQUEST)
...
-                $msg = sprintf('The controller must return a "Symfony\Component\HttpFoundation\Response" object but it returned %s.', $this->varToString($response));
-
-                // the user may have forgotten to return something
-                if (null === $response) {
-                    $msg .= ' Did you forget to add a return statement somewhere in your controller?';
-                }
-
-                throw new ControllerDoesNotReturnResponseException($msg, $controller, __FILE__, __LINE__ - 17);
+            if (!$event->hasResponse()) {
+                $message = sprintf(
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/HttpKernel.php]: please revert unrelated changes on this line

Comment (dimabory) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: done ☝

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @dimabory.
